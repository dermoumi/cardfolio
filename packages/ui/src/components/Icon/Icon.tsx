import type { FC } from "react";

import { ArrowLeft, CalendarClock, Network, Plus, Save, Trash, X } from "lucide-react";

const ICON_MAP = {
  network: Network,
  arrowLeft: ArrowLeft,
  calendarClock: CalendarClock,
  plus: Plus,
  x: X,
  trash: Trash,
  save: Save,
};

export type IconName = keyof typeof ICON_MAP;

export type IconProps = {
  name: IconName;
  label?: string;
  size?: number;
};

const Icon: FC<IconProps> = ({ name, label, size = 16 }) => {
  const IconComponent = ICON_MAP[name];

  return <IconComponent aria-label={label} size={size} />;
};

export default Icon;
