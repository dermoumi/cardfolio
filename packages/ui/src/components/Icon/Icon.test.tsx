import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import Icon from "./Icon";

describe("Icon component", () => {
  it("should render correctly", () => {
    const { container } = render(<Icon name="network" />);
    expect(container).toMatchSnapshot();
  });

  it("should add aria-label when provided", () => {
    const { container } = render(<Icon name="network" label="Network Icon" />);

    const iconElement = container.querySelector("svg");
    expect(iconElement?.ariaLabel).toBe("Network Icon");
  });
});
