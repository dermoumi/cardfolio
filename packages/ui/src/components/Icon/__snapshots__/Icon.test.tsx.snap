// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Icon component > should render correctly 1`] = `
<div>
  <svg
    class="lucide lucide-network"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      height="6"
      rx="1"
      width="6"
      x="16"
      y="16"
    />
    <rect
      height="6"
      rx="1"
      width="6"
      x="2"
      y="16"
    />
    <rect
      height="6"
      rx="1"
      width="6"
      x="9"
      y="2"
    />
    <path
      d="M5 16v-3a1 1 0 0 1 1-1h12a1 1 0 0 1 1 1v3"
    />
    <path
      d="M12 12V8"
    />
  </svg>
</div>
`;
